src/api/api_missions.nit

```diff
@@ -35,6 +36,40 @@ end
 
 class APIMission
 	super MissionHandler
+	super AuthHandler
+
+	redef fun post(req, res) do
+		var player = get_player(req, res)
+		if player == null then return
+		var mission = get_mission(req, res)
+		if mission == null then return
+		var post = req.post_args
+		if not post.has_key("source") then
+			res.error 400
+			print "No code provided for submission"
+			return
+		end
+		var code = post["source"]
+		if not post.has_key("lang") then
+			res.error 400
+			print "No lang provided for submission"
+			return
+		end
+		var engine = post["lang"]
+		if not config.engine_map.has_key(engine) then
+			res.error 400
+			print "Unknown language '{engine}'"
+			return
+		end
+		var runner = config.engine_map[engine]
+		var program = new Program(player, mission, code)
+		runner.run(program)
+
+		program.update_status(config)
+
+		res.json program.produce_summary
```

Comment:
I'm not sure how much information will be sent along with the `program` object, if the source is transmitted back wouldn't it be too large to send to a client?
If that's not the case, it could be done this way.